File: FrontEnd/src/app/features/reservations/shipCrews/classes/dtos/shipCrew-read-dto.ts
import { GenderAutoCompleteVM } from '../../../genders/classes/view-models/gender-autocomplete-vm'
import { Metadata } from 'src/app/shared/classes/metadata'
import { NationalityDropdownVM } from '../../../nationalities/classes/view-models/nationality-autocomplete-vm'
import { ShipAutoCompleteVM } from '../../../ships/classes/view-models/ship-autocomplete-vm'
import { CrewSpecialtyAutoCompleteVM } from '../../../crewSpecialties/classes/view-models/crewSpecialty-autocomplete-vm'

export interface ShipCrewReadDto extends Metadata {

    // PK
    id: number
    // Object fields
    gender: GenderAutoCompleteVM
    nationality: NationalityDropdownVM
    ship: ShipAutoCompleteVM
    specialty: CrewSpecialtyAutoCompleteVM
    // Fields
    lastname: string
    firstname: string
    birthdate: string
    passportNo: string
    passportExpiryDate: string
    isActive: boolean

}

File: API/Features/Reservations/ShipCrews/Dtos/ShipCrewReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations.ShipCrews {

    public class ShipCrewReadDto : IBaseEntity, IMetadata {

        // PK
        public int Id { get; set; }
        // Fields
        public string Lastname { get; set; }
        public string Firstname { get; set; }
        public string Birthdate { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }
        // Navigation
        public SimpleEntity Gender { get; set; }
        public SimpleEntity Nationality { get; set; }
        public SimpleEntity Ship { get; set; }

    }

}